internal/metrics/otelmetrics/counter.go

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package otelmetrics
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/metric"
+)
+
+type otelCounter struct {
+	counter  metric.Int64Counter
+	fixedCtx context.Context
+	option   metric.AddOption
+}
+
+func (c *otelCounter) Inc(value int64) {
+	c.counter.Add(c.fixedCtx, value, c.option)
```

Comment:
btw, passing `c.option` is what causes an extra memory allocation, if I remove it they drop to 0. It's not that passing the vararg by itself requires allocations (the vararg slice is allocated on the stack), but it's something that the SDK is doing when processing Add() with attributes.

Reply:
Yes, noticed the same thing. Anyways, should I proceed on implementing other metrics in the meantime?

Reply:
yes